metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py

```diff
@@ -1299,40 +1227,36 @@ def gen_database_containers(
             else None,
         )
 
-        self.stale_entity_removal_handler.add_entity_to_state(
-            type="container",
-            urn=make_container_urn(
-                guid=database_container_key.guid(),
-            ),
-        )
-
-        for wu in container_workunits:
-            self.report.report_workunit(wu)
-            yield wu
-
     def gen_schema_containers(
         self, schema: SnowflakeSchema, db_name: str
     ) -> Iterable[MetadataWorkUnit]:
-        domain_urn = self._gen_domain_urn(f"{db_name}.{schema.name}")
-
-        schema_container_key = self.gen_schema_key(
-            self.snowflake_identifier(db_name),
-            self.snowflake_identifier(schema.name),
+        schema_name = self.snowflake_identifier(schema.name)
+        database_container_key = gen_database_key(
+            database=self.snowflake_identifier(db_name),
+            platform=self.platform,
+            platform_instance=self.config.platform_instance,
+            env=self.config.env,
         )
 
-        database_container_key: Optional[PlatformKey] = None
-        if db_name is not None:
-            database_container_key = self.gen_database_key(
-                database=self.snowflake_identifier(db_name)
-            )
+        schema_container_key = gen_schema_key(
+            db_name=self.snowflake_identifier(db_name),
+            schema=schema_name,
+            platform=self.platform,
+            platform_instance=self.config.platform_instance,
+            env=self.config.env,
+        )
 
-        container_workunits = gen_containers(
-            container_key=schema_container_key,
+        yield from gen_schema_container(
             name=schema.name,
-            description=schema.comment,
+            schema=self.snowflake_identifier(schema.name),
+            database=self.snowflake_identifier(db_name),
```

Comment:
still feels a bit annoying that we need to provide the name twice, but i guess we can fix in a later PR